### ISSUE TITLE
 UserWarning: [Errno 2] No such file or directory on Solaris

### ISSUE THREAD

**Issue by mverrilli** [pydata/numexpr#146]
When importing pandas, I get the following warnings from numexpr on Solaris: 

> /opt/app/xxx/software/python/lib/python2.7/site-packages/numexpr/cpuinfo.py:50: UserWarning: [Errno 2] No such file or directory
  stacklevel=stacklevel+1)
> /opt/app/xxx/software/python/lib/python2.7/site-packages/numexpr/cpuinfo.py:71: UserWarning: [Errno 2] No such file or directory
  stacklevel=stacklevel+1):
> /opt/app/xxx/software/python/lib/python2.7/site-packages/numexpr/cpuinfo.py:57: UserWarning: [Errno 2] No such file or directory
  stacklevel=stacklevel+1)

> uname -a
> SunOS xxx 5.10 Generic_150400-04 sun4v sparc sun4v

Can I safely use numexpr on Solaris despite the warnings?

Thanks

**Comment by derickl** [comment]
I also have the same problem (numexpr 2.4)

/Applications/Kronos.app/Contents/Resources/py/lib/python2.7/site-packages/numexpr/cpuinfo.py:50: UserWarning: [Errno 2] No such file or directory
  stacklevel=stacklevel+1)
/Applications/Kronos.app/Contents/Resources/py/lib/python2.7/site-packages/numexpr/cpuinfo.py:71: UserWarning: [Errno 2] No such file or directory
  stacklevel=stacklevel+1):

**Comment by FrancescAlted** [comment]
This only happens when importing from pandas?  Can you send the proper sequence that raises the warning as well as pandas and numexpr versions that you are using?  Maybe this should be reported to pandas itself?

**Comment by grahamrjuk** [comment]
I am seeing this problem on Solaris simply running python and importing numexpr. No other actions needed. All tests succeed.

```
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Numexpr version:   2.4.4
NumPy version:     1.10.1
Python version:    2.7.1 (r271:86832, Dec  7 2010, 09:18:14) 
[GCC 3.4.3 (csl-sol210-3_4-branch+sol_rpath)]
Platform:          sunos5-sun4u
AMD/Intel CPU?     False
VML available?     False
Number of threads used by default: 8 (out of 8 detected cores)
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
<lots of dots>
----------------------------------------------------------------------
Ran 5436 tests in 18.517s

OK
<unittest.runner.TextTestResult run=5436 errors=0 failures=0>
```

Any code for numpy or numexpr was built with gcc 5.2.0. numpy and numexpr were installed using easy_install.

**Comment by grahamrjuk** [comment]
A quick look suggests several of the commands checked are not present on the Solaris machines:
`machine`, `sysconf`, `uname_i`, `sysctl`, `hw`

What is the impact of these being missing?

**Comment by FrancescAlted** [comment]
Well, provided that all tests pass, I would say that the warnings above should be pretty harmless, but patches (in this case for Solaris) are always welcome indeed.

**Comment by grahamrjuk** [comment]
I think I solved the problem - it was quite simple in the end. A couple of the commands were being provided incorrectly.
How best to return the patch to you? My local clone of numexpr is behind a firewall I am sure you won't be able to sync with.
There are only two lines to change.

**Comment by FrancescAlted** [comment]
That's great.  Could you send a [Pull Request](https://help.github.com/articles/using-pull-requests/)?  Thanks!

**Comment by FrancescAlted** [comment]
Solaris problems addressed in PR #189.  I am going to close this one, but if anyone else is still having issues, please feel free to re-open.

**Comment by gauss1986** [comment]
I had the same issue. All tests passed. I'm using Red Hat Enterprise Linux Server Version 7.2 (Maipo).

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Numexpr version:   2.6.2
NumPy version:     1.16.3
Python version:    3.6.3 |Anaconda, Inc.| (default, Oct 13 2017, 12:02:49) 
[GCC 7.2.0]
Platform:          linux-x86_64
AMD/Intel CPU?     True
VML available?     False
Number of threads used by default: 8 (out of 16 detected cores)
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Ran 5442 tests in 12.788s

OK

<unittest.runner.TextTestResult run=5442 errors=0 failures=0>
